core/src/App.html

```diff
@@ -1293,7 +1301,16 @@
           path = GenericHelpers.addLeadingSlash(path);
           openViewInNewTab(path);
         } else if (e.data.params.modal !== undefined) {
-          let path = buildPath(e.data.params, srcNode, srcPathParams);
+          let path;
+          const specialIframeMessageSource = IframeHelpers.getSpecialIframeMessageSource(
+            e,
+            specialIframeProps
+          );
+          if (specialIframeMessageSource.length > 0) {
+            path = buildPath(e.data.params, srcNode, srcPathParams);
+          } else {
+            path = buildPath(e.data.params, undefined, undefined);
```

Comment:
I prefer this way, which is cleaner. It should work as well.
```suggestion
            path = buildPath(e.data.params);
```